tests/phpunit/Application/QueryStringParserTest.php

```diff
@@ -49,6 +50,17 @@ public function testParsesNonExistenceConstraint(): void {
 		$this->assertTrue( $constraints->hasNoValue() );
 	}
 
+	public function testParsesInstance(): void {
+		$parser = new QueryStringParser();
+		$query = $parser->parse( 'haswbstatement:P42=Q68' );
+
+		$propertyId = new NumericPropertyId( 'P42' );
+		$itemId = new ItemId( 'Q68' );
+
+		$this->assertEquals( $propertyId, $query->getInstancePropertyId() );
+		$this->assertEquals( $itemId, $query->getInstanceItemId() );
+	}
```

Comment:
After this test is adjusted to provide `PropertyId $instanceOfId` to `new QueryStringParser`, some additional tests will be in order:

* testItemTypeIsNullWhenNotPresent
* testIgnoresHaswbstatementForNonInstanceOfIdProperties

and this test itself could be renamed to `testFindsFirstItemType` and use a query string like 

    unrelated haswbstatement:P42=Q68 haswbstatement:P42=Q69

so that the test verifies the precise behavior. For testIgnoresHaswbstatementForNonInstanceOfIdProperties you could have

    haswbstatement:P1=wrongId haswbstatement:P42=Q68 haswbstatement:P2=alsoWrong